drivers/modem/hl7800.c

```diff
@@ -2324,10 +2330,14 @@ static bool on_cmd_network_report_query(struct net_buf **buf, uint16_t len)
 		set_network_state(strtol(val, NULL, 0));
 
 		/* start work to adjust iface */
-		k_delayed_work_cancel(&ictx.iface_status_work);
-		k_delayed_work_submit_to_queue(&hl7800_workq,
-					       &ictx.iface_status_work,
-					       IFACE_WORK_DELAY);
+		rc = k_work_cancel_delayable(&ictx.iface_status_work);
```

Comment:
If you're going to unconditionally reschedule, there's no point in trying to cancel first.  Reschedule will replace any pending work.